### DISCUSSION TITLE
Issue adding search node to manager node

### DISCUSSION THREAD

**Discussion by noob2020-alt** [Security-Onion-Solutions/securityonion#10097] [Unsupported Versions]
Hello,

Build - security onion-2.3.220-20230301.iso
Platform - vmware ESXi 7
Manager Node - Centos 7 or Ubuntu 64
Search Node - Centos 7 or Ubuntu 64

Manager & Search on same network
Connect to UI via Hostname Ex. <managernode>

I am having an issue adding the search node to manager node through the installer.

NOTE: 
My DNS works fine
I can access the manager and search nodes from hostname and IP
Ex: Ping 192.168.1.1 - Good return
Ex: Ping <manager> and/or <search> - Good return

I can ping IP and hostname of manager from search node
I can SSH, from <search> into <manager> via IP but not hostname:
Ex: ssh user@192.168.1.1 - this returns a good result
Ex: ssh user@<manager> - This returns no route to host *****

After doing some light troubleshooting (I am not an expert) I found the following issue that I am not able to resolve:

1. During the installation of the <manager> everything works fine:
2. During installation of the search node, when the prompt to "Enter the server hostname: This is case sensitive", there is an issue that happens; I am not sure if this is normal or not. The issue:
3. <search> takes the entered name and adds it to the /etc/hosts/ file on the search node. 
4. After the installation fails and I am returned to the command line I remove the entry into the etc host file and am able to log into the <manager> node using the hostname. 
5. However, after reboot the etc/hosts file returns to normal - good config- and when I enter the new setup again and add the host name in the installer prompt I get the no route to host again. 
6. I am not sure on how to add the search node via ip address which would work since the <manager> node's hostname is automatically added to the search node's /etc/hosts file. 

Other methods attempted: 
remove entry in <search> node /etc/hosts file
deleted the following [ :: manager]

attempted to ssh user@managernode - success
rebooted and attempted to run setup again - same issue. 

Any help would be greatly appreciated.

**Comment by noob2020-alt** [comment]
I was able to figure this out. 

Whenever you attempt to add a search node to the manager node by inputing the hostname, the setup automatically places the hostname in the /etc/hosts file which causes an error of "no route to host" showing. 

Therefore; locking the etc/hosts file and rebooting did the trick
lock /etc/hosts command:    chattr +i /etc/hosts 
--- with this locked root can't even make changes to the /etc/hosts file, thus preventing the automatic hostname addition to /etc/hosts on the node you are trying to add.  

The unlock command (chattr -i /etc/hosts) is provided to make changes.

A suggestions would be to avoid initially starting during a distributed deployment and locking all the /etc/hosts files on the other nodes to prevent the above issues from occurring.

**Comment by cm-ops** [comment]
If you have your `sosetup.log` can you share what errors you are encountering? During setup the manager's hostname and IP are added to `/etc/hosts` as is a loopback for the node.

**Reply by TOoSmOotH** [reply]
Just to add on to this. I have seen something somewhat similar where my DNS is resolving the ipv6 address vs the ipv4. Do you by chance have ipv6 running? My issue is with unifi and the dns resolver.

**Reply by noob2020-alt** [reply]
Cutting off the IPv6 worked. Sorry for the long delay. Been offline for a minute. I shut off IPv6 and was able to get the prompt to enter theme. 
When I entered the correct hostname, I got the no route to host. SO I reset and entered the wrong name and was given the option to enter an IP. I entered the IP and it worked. Seems that the issue is with connecting via hostname. 

After running through setup it failed. Could be a dns issue, but I am able to ssh via the hostname from all other devices on the network.

**Reply by noob2020-alt** [reply]
This is the log I got when I tried to use the hostname. 

I may just do the install using an IP address. That one usually works without a hitch. 


2: ens33: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    
3: ens34: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
 
4: ens35: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000

5: ens36: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000

Beginning Security Onion iso install
Setup is running on TTY /dev/tty1
Kernel consoleblank value before: 600
Running 'setterm -blank 0' for TTY /dev/tty1
Kernel consoleblank value after: 0
2023-05-11T19:02:19Z | I | Disabling ipv6
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
Connection successfully activated (D-Bus active path: /org/freedesktop/NetworkManager/ActiveConnection/2)
Removing secmaster from /etc/hosts if present.
Adding secmaster to /etc/hosts with IP: ::
Generating SSH key
Removing old entry for manager from known_hosts if it exists
grep: /root/.ssh/known_hosts: No such file or directory
Copying the SSH key to the manager
/bin/ssh-copy-id: INFO: Source of key(s) to be installed: "/root/.ssh/so.key.pub"
ssh: connect to host secmaster port 22: Network is unreachable
ssh: connect to host secmaster port 22: Network is unreachable
Restarting Docker...
Could not determine version of Security Onion running on manager secmaster. Please check your network settings and run setup again.
2023-05-12T02:07:03Z | I | Installer version mismatch, downloading correct version from manager
ssh: connect to host secmaster port 22: Network is unreachable
Could not download .tar.gz from manager, please check your network settings and verify the file /opt/so/repo/.tar.gz exists on the manager.

**Comment by noob2020-alt** [comment]
<img width="754" alt="Screenshot 2023-05-11 at 9 27 16 PM" src="https://github.com/Security-Onion-Solutions/securityonion/assets/63116742/36351f33-32e6-40d2-b7c7-8cc32740c286">
